src/color.cc

```diff
@@ -213,6 +213,47 @@ ColorModule::~ColorModule()
 {
 }
 
+// Parses hex colors according to: https://drafts.csswg.org/css-color/#typedef-hex-color.
+// If the input is invalid, returns boost::none.
+// Supports the following formats:
+// * "#rrggbb"
+// * "#rrggbbaa"
+// * "#rgb"
+// * "#rgba"
+boost::optional<Color4f> parse_hex_color(const std::string& hex) {
+	// validate size. short syntax uses one hex digit per color channel instead of 2.
+	const bool short_syntax = hex.size() == 4 || hex.size() == 5;
+	const bool long_syntax = hex.size() == 7 || hex.size() == 9;
+	if (!short_syntax && !long_syntax) return boost::none;
+
+	if (hex[0] != '#') return boost::none;
+
+	// number of characters per color channel
+	const int stride = short_syntax ? 1 : 2;
+	const float channel_max = short_syntax ? 15 : 255;
+
+	std::array<int,4> rgba;
+	rgba[3] = channel_max; // default alpha channel to 100%
+
+	for (unsigned i = 0; i < (hex.size() - 1) / stride; i++) {
+		const std::string chunk = hex.substr(1 + i*stride, stride);
+
+		// validate - only [0-9a-zA-Z] are allowed
+		for (char c : chunk) {
+			if (!std::isdigit(c) && !('a' <= c && c <= 'z') && !('A' <= c && c <= 'Z')) return boost::none;
```

Comment:
Hexadecimal digits should be [0-9a-fA-F]

Reply:
Oops, just fixed it :)